libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/WayNameView.java

```diff
@@ -41,9 +49,10 @@ public String retrieveWayNameText() {
   }
 
   public void updateVisibility(boolean isVisible) {
```

Comment:
Isn't this method just a convenience for `View#setVisibility`? If so, both should have the same logic that checks for contents, preferably just one invokes another. If the view is made visible while there's no text, it should wait for the text to become available and show itself then.

Reply:
that's a good point. given that developer might only call the view's `setVisibility` instead of calling our `updateVisibility`. I will check it accordingly.

Reply:
I think this is missing:
> If the view is made visible while there's no text, it should wait for the text to become available and show itself then.

Otherwise, a user of this view is going to set the visibility and keep updating text, by the view will never appear.

TBH, I think the correct solution should be to keep the default behavior of `setVisibility` and make `!TextUtils.isEmpty(view.getWayNameText())` required by the user of the view before making it visible. That looks like the only way of avoiding undefined behavior. What do you think?

Reply:
You are basically capturing the same concern in https://github.com/mapbox/mapbox-navigation-android/pull/2547#discussion_r390467123 :)

It's hard to make the view show/hide itself while also not introducing undefined behavior for the user.

Reply:
keep it simple, just call the setVisibility and handle everything there. we have this `updateVisibility` method for user but since it's a view user has the choice to just call the `setVisibility` method.

Reply:
While it works for us, it's convenient, the logic itself might introduce undefined behavior.

Let's imagine that a user of the view calls `setVisibility(true)`, then immediately calls `getVisibility` and it returns `INVISIBLE`, which is confusing. Basically, the logic currently implemented in the `setVisibility(int)` should be handled by our presenter.